client/mysql.cc

```diff
@@ -5112,17 +5112,29 @@ static const char *construct_prompt()
           processed_prompt.append("unknown");
         break;
       case 'h':
+      case 'H':
       {
-	const char *prompt;
-	prompt= connected ? mysql_get_host_info(&mysql) : "not_connected";
-	if (strstr(prompt, "Localhost"))
-	  processed_prompt.append("localhost");
-	else
-	{
-	  const char *end=strcend(prompt,' ');
-	  processed_prompt.append(prompt, (uint) (end-prompt));
-	}
-	break;
+        const char *prompt;
+        prompt= connected ? mysql_get_host_info(&mysql) : "not_connected";
+        if (strstr(prompt, "Localhost"))
+        {
+          if (*c == 'h')
+            processed_prompt.append("localhost");
+          else
+          {
+            char hostname[HOST_NAME_MAX];
```

Comment:
I believe HOST_NAME_MAX can be undefined. Probably better to use something like FN_REFLEN?